server.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Setup:
+ *
+ * 1. Create an application in Asana.
+ * 2. Set the application's redirect URI to be:
+ *      `http://localhost:8338/oauth_callback`
+ * 3. Note the app's client ID and secret key for use when running the
+ *    web server, below.
+ *
+ * Usage:
+ *
+ *   export ASANA_CLIENT_ID=...
+ *   export ASANA_CLIENT_SECRET=...
+ *   [export PORT=...]
+ *   node oauth_webserver.js
+ */
+var Asana = require('asana');
```

Comment:
I'm curious on the inconsistencies here with `var` vs `const`?

Reply:
Most of this code is copy and paste from the Asana examples which are in js. Rather than deal with integrating it into the typescript and reworking the package.json scripts I opted for the quick and dirty solution of leaving the server as a js server that just serves the static build  items. I did not bother refactoring to const because without the ts compiler on-top there might be a risk that ES6 would not be supported by my/your node versions and this was already working.

I also went on a bit of a tangent to get asana auth working. Their docs about the auth flow are a little outdated. The frontend "implicit grant flow" is no longer supported so this was all added to allow a proper backend flow.

Reply:
In a production environment I would ideally move the `server.js` into `src/server.ts` and replace the `yarn start` script with one that will support the frontend and backend.

Reply:
Thanks for the answer and context!